### ISSUE TITLE
Accessing the SSM outputs of each layer

### ISSUE THREAD

**Issue by baturaysaglam** [state-spaces/mamba#208]
hello. not an issue but I'm wondering if it's possible to access/collect the outputs of the SSM module in layer's Mamba block?

**Comment by tridao** [comment]
Sure, it's just pytorch code, you can change it to get the SSM output.

**Comment by baturaysaglam** [comment]
thank you for your reply. I agree with you but I was wondering if there was a straightforward way to perform it. I delved into the code and have found [this function](https://github.com/state-spaces/mamba/blob/ce59daea3a090d011d6476c6e5b97f6d58ddad8b/mamba_ssm/modules/mamba_simple.py#L119). however, things could get really complicated if I modify it since it does not perform the SSM operation explicitly.  so, I don't want to break its functioning by mistake. 

instead, I've also noticed [another function](https://github.com/state-spaces/mamba/blob/ce59daea3a090d011d6476c6e5b97f6d58ddad8b/mamba_ssm/modules/mamba_simple.py#L208), from which I could obtain SSM logits directly. the code, however, does not enter to this function unless an ``inference_param`` is passed to the [forward function](https://github.com/state-spaces/mamba/blob/ce59daea3a090d011d6476c6e5b97f6d58ddad8b/mamba_ssm/modules/mamba_simple.py#L119). The only thing I want to do is obtaining the SSM logits in in-context learning, that is, get the corresponding outputs after a batch of prompts is passed, like inference. 

could you please help me with this case?

**Comment by ElfredTan** [comment]
same, have you solved this problem?
